### PR TITLE
Feature rules by user

### PR DESCRIPTION
**Pull request by maxkimambo**
+ Filtering displayed rules by the logged in User

### CODE REVIEW

File: client/src/app/components/add-rule/add-rule.component.html [hunk 1]
```diff
@@ -66,9 +66,12 @@ <h3>Parameters</h3>
                 <form #executionInterval="ngForm">
                     <h3>Run every</h3>
                     <mat-form-field class="form-element">
-                        <mat-label>Interval (minutes)</mat-label>
-                        <input [(ngModel)]="currentRule.executionInterval" matInput name="rule-interval" type="number"
-                            min="0" max="1440" step=15 required>
+                        <mat-label> Interval </mat-label>
```
Comment (paranerd) [inline client/src/app/components/add-rule/add-rule.component.html:69]: Remove spaces for consistency

File: client/src/app/components/rules/rules.component.ts [hunk 2]
```diff
@@ -69,19 +72,19 @@ export class RulesComponent implements OnInit {
...
   }
 
   /**
-   * Fetch all rules.
+   * Fetch all rules for logged in user.
    */
   loadRules() {
     this.http
-      .get<Array<Rule>>(`${environment.apiUrl}/rules`)
+      .get<Array<Rule>>(`${environment.apiUrl}/rules/user/${this.user?.id}`)
```
Comment (paranerd) [inline client/src/app/components/rules/rules.component.ts:87]: Why are we explicitly passing the user ID in the URL? The server should obtain this information from the authentication the user provides.

File: client/src/app/components/rules/rules.component.ts [hunk 3]
```diff
@@ -108,8 +111,9 @@ export class RulesComponent implements OnInit {
    * @param {Rule} rule
    */
   removeRule(rule: Rule) {
+    const userId = this.user!.id;
     this.http
-      .delete(`${environment.apiUrl}/rules/${this.user}/${rule.id}`)
+      .delete(`${environment.apiUrl}/rules/${userId}/${rule.id}`)
```
Comment (paranerd) [inline client/src/app/components/rules/rules.component.ts:116]: Same as above.

File: server/agents/source-agents/ambee/interfaces.ts [hunk 4]
```diff
@@ -72,9 +69,25 @@ export interface AgentMetadata {
 export interface DataPoint {
     dataPoint: string;
     name: string;
-    dataType: string | number | boolean | Date;
+    dataType: string | number | boolean | Date ;
```
Comment (paranerd) [inline server/agents/source-agents/ambee/interfaces.ts:72]: Is the last space auto-formatted? It looks like it doesn't belong there.

**Review by paranerd — CHANGES_REQUESTED** [review]

File: client/src/app/components/rules/rules.component.ts (hunk 2 above)
Comment (maxkimambo) [reply client/src/app/components/rules/rules.component.ts:87]: We previously discussed the need of passing user id to the api endpoint. 
This is required by design.

File: client/src/app/components/add-rule/add-rule.component.html (hunk 1 above)
Comment (paranerd) [reply client/src/app/components/add-rule/add-rule.component.html]: I meant the spaces around `Interval`. No other label has them and they shouldn't be there.

File: server/agents/source-agents/ambee/interfaces.ts (hunk 4 above)
Comment (paranerd) [reply server/agents/source-agents/ambee/interfaces.ts]: Space is still there. That is: the space between `Date` and the semicolon.

**Review by paranerd — CHANGES_REQUESTED** [review]

**Review by paranerd — APPROVED** [review]

**Comment by paranerd** [comment]
LGTM

**Comment by paranerd** [comment]
LGTM

**Pull request merged**